Runtime/Utility/CountdownTimer.cs

```diff
@@ -67,9 +69,12 @@ public virtual void Begin()
         /// </summary>
         public virtual void Cancel()
         {
-            IsRunning = false;
             CancelInvoke(nameof(Complete));
-            Cancelled?.Invoke();
+            if (IsRunning)
+            {
+                Cancelled?.Invoke();
+            }
+            IsRunning = false;

```

Comment:
Move this into the if to save an assignment that doesn't do anything as it's already false.